lib/direct_uploader/model.rb

```diff
@@ -6,15 +6,26 @@ module Model
 
     included do
       before_save :save_direct_uploads
-      cattr_accessor :direct_uploader_fields
-      cattr_accessor :direct_uploader_field_options
+      class_attribute :direct_uploader_fields
+      class_attribute :direct_uploader_field_options
       attr_writer :direct_uploader_adapter
+
+      delegate :direct_uploader_field_options, to: :class
     end
 
     module ClassMethods
+      # Note class_attribute works fine with immutable objects like String, but not with mutables like Hash
+      # The trick is to use the inherited method in the parent class to clone the attribute in subclass
+      # https://stackoverflow.com/questions/28041368/proper-way-to-use-class-attribute-with-hash
+      def inherited(child_class)
+        super
+        child_class.direct_uploader_fields = self.direct_uploader_fields.clone
+        child_class.direct_uploader_field_options = self.direct_uploader_field_options.clone
+      end
```

Comment:
le super est very important, car on surcharge le comportement du STI de rails (et on se prend un beau `rails undefined method 'synchronize' for nil` inextricable)